src/table_properties.rs

```diff
@@ -77,61 +72,50 @@ impl<'a> TablePropertiesCollectionIter<'a> {
             }
         }
     }
+}
 
-    pub fn valid(&self) -> bool {
-        unsafe { crocksdb_ffi::crocksdb_table_properties_collection_iter_valid(self.inner) }
-    }
+impl<'a> Iterator for TablePropertiesCollectionIter<'a> {
+    type Item = (&'a str, &'a TableProperties);
 
-    pub fn next(&mut self) {
+    fn next(&mut self) -> Option<(&'a str, &'a TableProperties)> {
         unsafe {
-            crocksdb_ffi::crocksdb_table_properties_collection_iter_next(self.inner);
-        }
-    }
+            if !crocksdb_ffi::crocksdb_table_properties_collection_iter_valid(self.inner) {
+                return None;
+            }
 
-    pub fn key(&self) -> &'a str {
-        unsafe {
             let mut klen: size_t = 0;
             let k = crocksdb_ffi::crocksdb_table_properties_collection_iter_key(self.inner,
                                                                                 &mut klen);
             let bytes = slice::from_raw_parts(k, klen);
-            str::from_utf8(bytes).unwrap()
-        }
-    }
-
-    pub fn value(&self) -> TableProperties {
-        unsafe {
-            let props = TableProperties::new();
-            crocksdb_ffi::crocksdb_table_properties_collection_iter_value(self.inner, props.inner);
-            props
+            let key = str::from_utf8(bytes).unwrap();
+            let props = crocksdb_ffi::crocksdb_table_properties_collection_iter_value(self.inner);
+            let val = TableProperties::from_ptr(props);
+            crocksdb_ffi::crocksdb_table_properties_collection_iter_next(self.inner);
+            Some((key, val))
         }
     }
 }
 
 pub struct TableProperties {
-    inner: *mut DBTableProperties,
+    inner: DBTableProperties,
 }
 
-impl Drop for TableProperties {
-    fn drop(&mut self) {
+impl TableProperties {
+    fn from_ptr<'a>(ptr: *const DBTableProperties) -> &'a TableProperties {
```

Comment:
Can we return a `TableProperties<'a>` and keep the inner pointer here?
It's a bit of magic for me to figure out how this work.